content/custom_resources/glossary.md

```diff
@@ -0,0 +1,607 @@
+
+Use to ensure a property value is of a particular ruby class, such as:
+
+- `true`
+- `false`
+- `nil`
+- `String`
+- `Array`
+- `Hash`
+- `Integer`
+- `Symbol`
+
+Use an array of Ruby classes to allow a value to be of more than one type. For example:
+
+```ruby
+property :aaaa, String
+property :bbbb, Integer
+property :cccc, Hash
+property :dddd, [true, false]
+property :eeee, [String, nil]
+property :ffff, [Class, String, Symbol]
+property :gggg, [Array, Hash]
+```
+
+## sensitive
+
+Introduced: Chef Infra Client 12.14
+
+A property can be marked sensitive by specifying `sensitive: true` on
+the property. This prevents the contents of the property from being
+exported to data collection and sent to an Automate server.
+
+## validators
+
+{{ dsl/property_validation_parameter }}
+
+## desired_state
+
+Add `desired_state:` to set the desired state property for a resource.
+
+| Allowed values | Default |
+| -------------- | ------- |
+| `true` `false` | `true`  |
+
+- When `true`, the state of the property is determined by the state of
+  the system
+- When `false`, the value of the property impacts how the resource
+  executes, but it is not determined by the state of the system.
+
+For example, if you were to write a resource to create volumes on a
+cloud provider you would need define properties such as `volume_name`,
+`volume_size`, and `volume_region`. The state of these properties would
+determine if your resource needed to converge or not. For the resource
+to function you would also need to define properties such as
+`cloud_login` and `cloud_password`. These are necessary properties for
+interacting with the cloud provider, but their state has no impact on
+decision to converge the resource or not, so you would set
+`desired_state` to `false` for these properties.
+
+```ruby
+property :volume_name, String
+property :volume_size, Integer
+property :volume_region, String
+property :cloud_login, String, desired_state: false
+property :cloud_password, String, desired_state: false
+```
+
+## identity
+
+{{< note >}}
+
+The `identity` setting is kept for backwards compatability. It is no longer used.
+
+{{< /note >}}
+
+The identity setting is used for reporting purposes.
+
+- When `true`, data for that property is returned as part of the resource data set and may be available to external applications, such as reporting
+- When `false`, no data for that property is returned.
+
+If no properties are marked `true`, the property that defaults to the `name` of the resource is marked `true`.
+
+For example, the following properties define `username` and `password` properties with no default values specified, but with `identity` set to `true` for the user name:
+
+```ruby
+provides :user
+
+property :username, String, identity: true
+property :password, String
+```
+
+```ruby
+user 'chef' do
+  password '12345'
+end
+```
+
+## run_context
+
+Chef loads and tracks the current run in the run context object.
```

Comment:
We need a better description of this

Reply:
Yeah, I could really do with jumping on a call or chat to discuss this one. I've implemented in both HAProxy and Samba.I know the how, but not the *why*. 

For example I know we need to force things into the `:root` run_context for[ multiple resources to find each other. ](https://github.com/sous-chefs/samba/blob/c1f14f2f81d7f78dafcef44e1abfc6024dca4b1f/resources/server.rb#L147-L151), but I don't know how the run_context object is structured.